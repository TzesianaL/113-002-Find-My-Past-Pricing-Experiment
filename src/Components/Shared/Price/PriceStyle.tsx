import styled from 'styled-components'

/**
 * Styled components are used in order to style Price component. Props are passed through ThemeProvider.
 */
export const PriceStyle = styled.div`
  display: flex;
  flex-direction: row;
  color: ${(props) => props.theme.color || props.theme.colors?.primary};
  font-size: ${(props) =>
    props.theme.priceFontSize || props.theme.fontSize || 1.2}rem;
  font-family: BoldVf;
  margin: 0em 0em 0.5em 0em;
  del {
    color: red;
  }
  .pricePeriodPlaceHolder {
    font-size: small;
    color: white;
    line-height: 23px;
  }
`
