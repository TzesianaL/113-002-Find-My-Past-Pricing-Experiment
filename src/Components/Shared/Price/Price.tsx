import React, { FC } from 'react'
import { getPeriodText } from 'functions/functions'
import { Periods } from 'types'
import { getSelectedCondition } from 'redux/reducers/dataStore/dataStore.selectors'
import { useSelector } from 'react-redux'
import { getCost } from 'redux/reducers/logicStore/logicStore.selectors'
import { CSSProperties } from 'styled-components'
import { PriceStyle } from './PriceStyle'

export interface PriceProps {
  /**
   * String variable stores the price and currency
   */
  price: number
  currency?: string
  priceString?: string
  newPrice?: number
  pricePeriod?: Periods
  style?: CSSProperties
  id?: string
}

/**
 * Price component is used to display the price and currency.
 * PriceStyle is used to style the component.
 */
const Price: FC<PriceProps> = ({
  price,
  currency = 'GBP',
  priceString = 'month',
  newPrice,
  pricePeriod,
  style,
  id,
}) => {
  const cost = useSelector(getCost)

  const selectedCondition = useSelector(getSelectedCondition)

  // let priceToShow = '0'

  // if (price !== undefined) {
  //   priceToShow = new Intl.NumberFormat('en-US', {
  //     style: 'currency',
  //     currency,
  //   }).format(price)
  // }

  // if (price === undefined && priceString === undefined) {
  //   throw Error('Price component was used without any price')
  // }

  if (priceString === undefined && price !== undefined && price !== null) {
    const oldPriceOutput = new Intl.NumberFormat('en-US', {
      style: 'currency',
      currency,
    }).format(price)
    if (newPrice !== undefined && newPrice !== null) {
      const newPriceOutput = new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency,
      }).format(newPrice)

      return (
        <PriceStyle style={style}>
          <del>{oldPriceOutput}</del>- Now {newPriceOutput}
          {pricePeriod && ` ${getPeriodText(pricePeriod)}`}
        </PriceStyle>
      )
    }

    return (
      <PriceStyle style={style}>
        {oldPriceOutput}
        {pricePeriod && ` ${getPeriodText(pricePeriod)}`}
      </PriceStyle>
    )
  }

  return (
    <PriceStyle style={style}>
      <strong>Â£{price.toFixed()}</strong> &nbsp;{' '}
      <span className="pricePeriodPlaceHolder">
        {' '}
        {pricePeriod && ` ${getPeriodText(pricePeriod)}`}
      </span>
      &nbsp;
    </PriceStyle>
  )
}

export default Price
