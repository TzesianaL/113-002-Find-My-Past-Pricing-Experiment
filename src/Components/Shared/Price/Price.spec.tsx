import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import Price from './Price'

describe('<Price />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Price price={100} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should display £100', () => {
    const el = render(<Price price={100} />)
    expect(el.getByText('£100.00')).toBeInTheDocument()
  })
})
