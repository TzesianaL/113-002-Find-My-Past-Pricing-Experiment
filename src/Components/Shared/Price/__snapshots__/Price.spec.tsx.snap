// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1.2rem;
  font-family: BoldVf;
  margin: 0em 0em 0.5em 0em;
}

.c0 del {
  color: red;
}

.c0 .pricePeriodPlaceHolder {
  font-size: small;
  color: grey;
  line-height: 23px;
}

<div
  className="c0"
>
  <span
    className="pricePeriodPlaceHolder"
  >
     
  </span>
    
  <strong>
    £100.00
  </strong>
    
   
  <span
    className="pricePeriodPlaceHolder"
  >
    a month
  </span>
   
</div>
`;
